glue_jupyter/bqplot/histogram/reacthistogram.py

```diff
@@ -0,0 +1,173 @@
+from glue.core.subset import roi_to_subset_state
+from glue.core.roi import RangeROI
+
+from glue.viewers.histogram.state import HistogramLayerState
+from glue.viewers.histogram.state import HistogramViewerState
+from glue_jupyter.common.state_widgets.layer_histogram import HistogramLayerStateWidget
+from glue_jupyter.common.state_widgets.viewer_histogram import (
+    HistogramViewerStateWidget,
+)
+
+from glue.core.exceptions import IncompatibleAttribute
+from glue.utils import color2hex
+
+
+from ..common.reactviewer import BqplotBaseViewReact
+from ...common.hooks import use_echo_state
+
+import react_ipywidgets as react
+import react_ipywidgets.bqplot as bq
+import react_ipywidgets.logging as logging
+
+
+import numpy as np
+
+
+def calculate_bins(
+    viewer_state: HistogramViewerState, layer_state: HistogramLayerState
+):
+    use_echo_state(viewer_state, "hist_x_max")
```

Comment:
What effect does this line have in practice as no return value is being used?

Reply:
Yes, that is the 'magic' from react, combined with having 2 places to store the state. The idea is that if you use `use_echo_state`, your 'render function' (i.e. the function wrapped in @react.component) will be re-executed. Typically, you'd use the return value, but they are also directly accessible using e.g. `viewer_state.normalize` etc.